src/app/app-routing.module.ts

```diff
@@ -50,6 +52,13 @@ const routes: Routes = [
       project: ProjectDetailResolver
     },
   },
+  {
+    path: 'p/:code/commentperiod/:id',
+    component: CommentPeriodComponent,
+    resolve: {
+      project: CommentPeriodResolver
```

Comment:
Please rename to `commentPeriod`. We are resolving a comment period here, not a project...